vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/Grid.java

```diff
@@ -2420,15 +2420,7 @@ private int getLastFooterLayerIndex() {
         return -1;
     }
 
-    /**
-     * {@inheritDoc}
-     *
-     * @deprecated use instead one of the {@code setItems} methods which provide
-     *             access to either {@link GridListDataView} or
-     *             {@link GridLazyDataView}
-     */
-    @Deprecated
-    public void setDataProvider(DataProvider<T, ?> dataProvider) {
+    protected void setDataProvider(DataProvider<T, ?> dataProvider) {
```

Comment:
I would suggest to keep the access modifier public in Grid:
if one wants to pass a data provider to Grid, he/she can use `setItems(DataProvider<T, Void> dataProvider)`, but the problem with it is the Void type, which prevents to pass, for example, `DataProvider<Person, String>`.

So in general we can keep it deprecated, but explicitly say in the javadoc in what cases this method can be used: when having a data provider with filter that cannot be transformed to `DataProvider<T, Void>`.

Changing the generic type `Void` to something more clever doesn't seem an easy task and would need a re-design.

Reply:
Ok, it will solve the issue with `CrudGrid`.